lib/user/widgets/user_selector.dart

```diff
@@ -71,6 +87,44 @@ class _UserSelectorState extends State<UserSelector> {
                 }
                 widget.onCommunityChanged!(resolvedCommunity);
               }
+
+              // If there is a selected post, see if we can resolve it to the new user's instance.
+              if (widget.postActorId?.isNotEmpty == true && widget.onPostChanged != null) {
+                PostView? resolvedPost;
+                try {
+                  final ResolveObjectResponse resolveObjectResponse = await LemmyApiV3(newUser.instance!).run(ResolveObject(q: widget.postActorId!));
+                  resolvedPost = resolveObjectResponse.post;
+                  if (resolvedPost != null) {
+                    widget.onPostChanged!(resolvedPost);

```

Comment:
Return early here so that we don't have to go through the rest of the logic if it succeeds!

Reply:
I'm not sure I completely understand what you want me to change here. Do you mean, if we're able to successfully resolve the post, for example, to skip the rest of the object resolution?

I guess that could work in this context, but maybe `UserSelector` could be used in a different context where we'd want multiple objects resolved, so we can't know (internally) that we're done after one succeeds. What do you think?

Reply:
Oops, I forgot to respond back to this! I was under the impression that we only had to resolve a single object here, but you do make a good point that we might need to resolve multiple objects in the future!

I'll resolve this issue for now.